### ISSUE TITLE
compatibility with angular-material

### ISSUE THREAD

**Issue by seiyria** [vitalets/angular-xeditable#232]
https://github.com/angular/material -- it would be nice to use x-editable with this! currently, I can get the buttons, and probably the input, but there's just a lot to do.

However, it might want to wait until the select component is implemented over there.

**Comment by Toilal** [comment]
:+1:

**Comment by Toilal** [comment]
I'm not sure it is that hard. It seems to be possible by creating a new `angular-material` [theme](https://github.com/vitalets/angular-xeditable/blob/master/src/js/themes.js).

**Comment by Toilal** [comment]
@seiyria, this seems to work nicely :

```js
    'angular-material': {
      formTpl:      '<form class="editable-wrap"></form>',
      noformTpl:    '<span class="editable-wrap"></span>',
      controlsTpl:  '<md-input-container class="editable-controls"></md-input-container>',
      inputTpl:     '',
      errorTpl:     '<div ng-messages="{message: $error}"><div class="editable-error" ng-message="message">{{$error}}</div></div>',
      buttonsTpl:   '<span class="editable-buttons"></span>',
      submitTpl:    '<md-button type="submit">save</md-button>',
      cancelTpl:    '<md-button type="button" ng-click="$form.$cancel()">cancel</md-button>'
    },
```

**Comment by seiyria** [comment]
@Toilal That worked ok, I had to make a few adjustments (submitTpl needed .md-primary, cancelTpl needed .md-warn) but other than that it was fine. It would still be nice to get this upstream though.

**Comment by Toilal** [comment]
Problem is that there's no maintainer ... @vitalets @assisrafael Could you give some people access rights to maintain this repo ? There are many PRs pending ...

**Comment by seiyria** [comment]
Another thing about this. Now that angular/material has updated, their templates have changed. The structure is a LOT different, and I wish a-xeditable was more flexible in this regard.

Basically, instead of giving it stuff like "inputTpl" I would rather just give it an entire template, structured however I want. It should be up to me to ensure the classes, etc get added correctly. In its current state, though, it's unusable AFAICT.

**Comment by kevohagan** [comment]
@seiyria @Toilal  did you guys manage to get this working with angular material at the end? :)

**Comment by Toilal** [comment]
Yes, with the snippet i've pasted, but I don't use it now. Don't know if it still works.

**Comment by kevohagan** [comment]
@Toilal its not working for me... if i manage to figure this out ill post it here :)

**Comment by pdonorio** [comment]
+1

This plugin is very great and used a lot in my previous project, thanks for your great work.
I am trying material design now that the official angular material has come to 1.0 version but still no luck. @Toilal did you manage to solve the snippet? May i help in some way?

**Comment by mostafabarmshory** [comment]
+1 It is nice.

**Comment by avalonabecker** [comment]
I was having trouble with Toilal's setup for the save button. It was not visible, because I'm using the angular-material theme "blue-grey". This is what worked for me:
.config($mdThemingProvider){
     $mdThemingProvider.theme('default').primaryPalette('blue-grey').accentPalette('red');
}

angular.module('myApp').run(function(editableOptions, editableThemes) {
    editableThemes['angular-material'] = {
        formTpl:      '<form class="editable-wrap"></form>',
        noformTpl:    '<span class="editable-wrap"></span>',
        controlsTpl:  '<md-input-container class="editable-controls" ng-class="{\'md-input-invalid\': $error}"></md-input-container>',
        inputTpl:     '',
        errorTpl:     '<div ng-messages="{message: $error}"><div class="editable-error" ng-message="message">{{$error}}</div></div>',
        buttonsTpl:   '<span class="editable-buttons"></span>',
        submitTpl:    '<md-button type="submit" class="md-icon-button"><i class="material-icons">&#xE876;</i></md-button>',
        cancelTpl:    '<md-button type="button" class="md-icon-button" ng-click="$form.$cancel()"><i class="material-icons">&#xE5C9;</i></md-button>'
    };

    editableOptions.theme = 'angular-material';
  };
});

**Comment by OscarAgreda** [comment]
@Toilal 
Thank you , your addition to angular 

> app.run

 works great . 
I am using angular material 1.1.0 - and no bootstrap.css whatsoever 

<p>i added the following css tweaks:</p>
<pre>
md-input-container.editable-controls {
    padding: 1px !important;
    padding-left: 10px !important;
    margin: 1px !important;
}
.editable-input.md-input {
    font-size: 18px !important;
    font-weight: 600;
    color: #716e6e;
    max-width: 100px !important;
    margin: 1px !important;
}
.editable-wrap.editable-text {
    margin: 1px !important;
}
md-input-container.editable-controls [ng-message] {
    position: static !important;
}
</pre>

**Comment by harshaldhake** [comment]
@OscarAgreda : can u please advise exactly where did u add these style? Also is there any code things added as per @Toilal

**Comment by gerome0123** [comment]
can u give a working jsfiddle  or plunker for this thanks guys

**Comment by gerome0123** [comment]
@OscarAgreda can you give an example running exeditable with angular material T_T

**Comment by Scheinin** [comment]
@Toilal  I use that in md-data-table, but can not cover the original data.Do you encounter this situation?
![image](https://user-images.githubusercontent.com/26054233/35133628-39d1765e-fd0c-11e7-9a1a-fe30cec88c0f.png)

**Comment by ikwijaya** [comment]
error : <md-input-container> can only have *one* <input>, <textarea> or <md-select> child element!
when using editable-radiolist.
